### PR TITLE
Add documentation for GET_HOST_CATEGORIES function

### PR DESCRIPTION
**Pull request by nrllh**
This pull request adds documentation for the `GET_HOST_CATEGORIES `function, including its description, usage, and methodology.

### CODE REVIEW

File: src/content/docs/reference/functions/get_host_categories.mdx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  `httparchive.urls.latest_crux_mobile` AS r,
+  UNNEST(httparchive.fn.GET_HOST_CATEGORIES(r.url)) AS c;
+  ```
+  
+  </TabItem>
+</Tabs>
+
+
+## Methodology
+The classification of hostnames is performed using Chrome's Topics API (currently `chrome5`)  by [Nurullah Demir](https://ndemir.com) and [Yohan Beugin](https://yohan.beugin.org/) using [this repository](https://github.com/yohhaan/httparchive-topics-classification). The taxonomy used is version 2, which consists of 469 categories as defined in the [Topics API taxonomy v2](https://github.com/patcg-individual-drafts/topics/blob/a116e9e404dd96c7793fe6542c53e9a9d93cb75b/taxonomy_v2.md).
```
Comment (rviscomi) [inline src/content/docs/reference/functions/get_host_categories.mdx:84]: Could we include more info about how recently the `chrome5` model was trained?

File: src/content/docs/reference/functions/get_host_categories.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+The function specifically addresses domains (currently top 50) known for having numerous hashed subdomains. Some of these domains include:
+- `googlesyndication.com`
+- `gstatic.com`
+- `cloudfront.net`
+- `akamaihd.net`
+- `doubleclick.net`
+- `amazonaws.com`
+
+And others listed and discussed in [this issue](https://github.com/HTTPArchive/httparchive.org/issues/868).
```
Comment (rviscomi) [inline src/content/docs/reference/functions/get_host_categories.mdx:108]: Let's link to the function as the source of truth
```suggestion
And others as seen in the [function source](https://console.cloud.google.com/bigquery?ws=!1m5!1m4!6m3!1shttparchive!2sfn!3sGET_HOST_CATEGORIES).
```

File: src/content/docs/reference/functions/get_host_categories.mdx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+title: GET_HOST_CATEGORIES function
+description: Reference docs for the httparchive.fn.GET_HOST_CATEGORIES function
+---
+
+# `GET_HOST_CATEGORIES` function
+
+## Description
+We are happy to release the largest open-source dataset of website categories, featuring 147 million hosts and 31 million domains, making it the most extensive open-source data available in this area.
+
+The `GET_HOST_CATEGORIES` function retrieves the category information of a given hostname. It maps hostnames to their respective categories using a pre-classified dataset.
```
Comment (rviscomi) [inline src/content/docs/reference/functions/get_host_categories.mdx:11]:
```suggestion
The [`GET_HOST_CATEGORIES`](https://console.cloud.google.com/bigquery?ws=!1m5!1m4!6m3!1shttparchive!2sfn!3sGET_HOST_CATEGORIES) function retrieves the category information of a given hostname. It maps hostnames to their respective categories using a pre-classified dataset. 
```

**Review by rviscomi — CHANGES_REQUESTED** [review]
Thanks so much for documenting this! Just a few small comments.

File: src/content/docs/reference/functions/get_host_categories.mdx [hunk 4]
```diff
@@ -0,0 +1,109 @@
+---
+title: GET_HOST_CATEGORIES function
+description: Reference docs for the httparchive.fn.GET_HOST_CATEGORIES function
+---
+
+# `GET_HOST_CATEGORIES` function
+
+## Description
+We are happy to release the largest open-source dataset of website categories, featuring 147 million hosts and 31 million domains, making it the most extensive open-source data available in this area.
```
[the remark this answers is not in the document]
Comment (nrllh) [reply src/content/docs/reference/functions/get_host_categories.mdx:8]: I was also not sure about that, feel free to remove this sentence

File: src/content/docs/reference/functions/get_host_categories.mdx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+This model is applied to all requests in the HTTPArchive's dataset, including not only the visited origins but also the seen requests during the crawl up until June 2024. 
+
+### Raw Data
+The raw data for the classifications is stored in the `httparchive.urls.categories` table. This table consists of pre-classified hostnames with their corresponding categories. The categories follow a hierarchical structure, providing both specific subcategories and broader parent categories.
+
+Please consider the limitations of our method regarding some hosts discussed [here](https://github.com/HTTPArchive/httparchive.org/issues/868). Thus, while this data can be accessed directly, we highly recommend using the `GET_HOST_CATEGORIES` function due to the handling of hashed subdomains. If your analysis requires working with domains (e.g., `google.com` instead of `maps.google.com`), accessing the raw data directly is also appropriate.
```
[the remark this answers is not in the document]
Comment (nrllh) [reply src/content/docs/reference/functions/get_host_categories.mdx:91]: I think linking to the issue makes sense for anyone interested in discovering the limitations more

File: src/content/docs/reference/functions/get_host_categories.mdx (hunk 1 above)
Comment (nrllh) [reply src/content/docs/reference/functions/get_host_categories.mdx:84]: Yes, it makes sense. It's currently the latest model shipped in Chrome for topics.

**Comment by rviscomi** [comment]
Thanks, I'll merge and open follow-up PRs for any updated.

**Review by rviscomi — APPROVED** [review]

**Pull request merged**